$MirOS: ports/sysutils/e2fsprogs/patches/patch-lib_blkid_probe_c,v 1.1.7.1 2005/03/18 15:52:04 tg Exp $
--- lib/blkid/probe.c.orig	Sat May  7 18:34:53 2005
+++ lib/blkid/probe.c	Tue Jan  3 02:54:29 2006
@@ -571,8 +571,8 @@ blkid_dev blkid_verify(blkid_cache cache
 		return dev;
 
 	DBG(DEBUG_PROBE,
-	    printf("need to revalidate %s (time since last check %lu)\n", 
-		   dev->bid_name, diff));
+	    printf("need to revalidate %s (time since last check %lld)\n", 
+		   dev->bid_name, (long long)diff));
 
 	if (((fd = open(dev->bid_name, O_RDONLY)) < 0) ||
 	    (fstat(fd, &st) < 0)) {
@@ -666,7 +666,7 @@ found_type:
 		blkid_set_tag(dev, "TYPE", type, 0);
 				
 		DBG(DEBUG_PROBE, printf("%s: devno 0x%04llx, type %s\n",
-			   dev->bid_name, st.st_rdev, type));
+			   dev->bid_name, (unsigned long long)st.st_rdev, type));
 	}
 
 	close(fd);
